### ISSUE TITLE
Unhandled exception: System.ArgumentException: '24' is not a valid version string.

### ISSUE THREAD

**Issue by steverandy** [electron-userland/electron-builder#1999]
* **Version**: 19.24.1
`electron-builder-squirrel-windows`: 19.24.0

* **Target**: win

App version is set to `1.0.0-beta.24`.

Got unhandled exception during build process for Windows.
Build was successful, if downgraded to v19.16.0.

I think it's related to these changes: https://github.com/electron-userland/electron-builder/commit/f63086d13558771508314a9620dbbdf0613f1210#diff-5d9edd85053aca39cb001f43cdd633e4R81

```
2017-08-27 13:13:25> Unhandled exception: System.ArgumentException: '24' is not a valid version string.
Parameter name: version
  at NuGet.SemanticVersion.Parse (System.String version) [0x0003b] in <259501d16aec4eda8604f04901763545>:0 
  at NuGet.SemanticVersion..ctor (System.String version) [0x00000] in <259501d16aec4eda8604f04901763545>:0 
  at Squirrel.VersionExtensions.ToSemanticVersion (System.String fileName) [0x00021] in <259501d16aec4eda8604f04901763545>:0 
  at Squirrel.ReleaseEntry.get_Version () [0x00006] in <259501d16aec4eda8604f04901763545>:0 
  at Squirrel.ReleaseEntry+<>c.<WriteReleaseFile>b__44_0 (Squirrel.ReleaseEntry x) [0x00000] in <259501d16aec4eda8604f04901763545>:0 
  at System.Linq.EnumerableSorter`2[TElement,TKey].ComputeKeys (TElement[] elements, System.Int32 count) [0x00010] in <b462790788ef4194924839a1c3047e68>:0 
  at System.Linq.EnumerableSorter`1[TElement].ComputeMap (TElement[] elements, System.Int32 count) [0x00000] in <b462790788ef4194924839a1c3047e68>:0 
  at System.Linq.EnumerableSorter`1[TElement].Sort (TElement[] elements, System.Int32 count) [0x00000] in <b462790788ef4194924839a1c3047e68>:0 
  at System.Linq.OrderedEnumerable`1[TElement].SortedMap (System.Linq.Buffer`1[TElement] buffer) [0x00006] in <b462790788ef4194924839a1c3047e68>:0 
  at System.Linq.OrderedEnumerable`1+<GetEnumerator>d__3[TElement].MoveNext () [0x0003d] in <b462790788ef4194924839a1c3047e68>:0 
  at System.Linq.Enumerable+SelectIPartitionIterator`2[TSource,TResult].MoveNext () [0x00029] in <b462790788ef4194924839a1c3047e68>:0 
  at System.String.Join (System.String separator, System.Collections.Generic.IEnumerable`1[T] values) [0x0001f] in <ab2f5ad0f8c84dfe97dfd6a06a64cf44>:0 
  at Squirrel.ReleaseEntry.WriteReleaseFile (System.Collections.Generic.IEnumerable`1[T] releaseEntries, System.IO.Stream stream) [0x0007f] in <259501d16aec4eda8604f04901763545>:0 
  at Squirrel.ReleaseEntry.WriteReleaseFile (System.Collections.Generic.IEnumerable`1[T] releaseEntries, System.String path) [0x0000a] in <259501d16aec4eda8604f04901763545>:0 
  at Squirrel.Update.Program.ReleasifyElectron (System.String package, System.String targetDir, System.String baseUrl) [0x00166] in <259501d16aec4eda8604f04901763545>:0 
  at Squirrel.Update.Program.executeCommandLine (System.String[] args) [0x00452] in <259501d16aec4eda8604f04901763545>:0 
  at Squirrel.Update.Program.main (System.String[] args) [0x00097] in <259501d16aec4eda8604f04901763545>:0 
```

**Comment by VerteDinde** [comment]
@steverandy I'm having this problem as well - any luck solving it?

Versions:     
 - electron-builder: 19.27.7 & 19.12,
 - electron-builder-squirrel-windows: 19.28.3

Target: win, squirrel

```
2017-09-14 03:02:46> Unhandled exception: System.ArgumentException: '0.0.9-hikers-guide' is not a valid version string.
Parameter name: version
   at NuGet.SemanticVersion.Parse(String version)
   at NuGet.ManifestMetadata.NuGet.IPackageName.get_Version()
   at NuGet.LocalPackage.ReadManifest(Stream manifestStream)
   at NuGet.ZipPackage.EnsureManifest()
   at NuGet.ZipPackage..ctor(String filePath, Boolean enableCaching)
   at Squirrel.Update.Program.ReleasifyElectron(String package, String targetDir, String baseUrl (etc, etc)
```

**Comment by steverandy** [comment]
@VerteDinde I have not found a solution so far. I'm stuck with v19.16.0.

**Comment by VerteDinde** [comment]
@steverandy Bummer - I'll keep trying on my end and will let you know if I solve it/how. Looks like it's being discussed in Nuget as well. https://github.com/NuGet/Home/issues/3070

**Comment by VerteDinde** [comment]
Okay, I think I found something semi-related: Looks like this is a known problem in NuGet because they don't support SemVer 2.0 Relevant link here: https://github.com/OctopusDeploy/Issues/issues/2260

> Turns out, it is a valid string, but only for SemVer 2.0. NuGet only supports SemVer 1.0, which does not allow .-separated pre-release identifiers, which is a major bummer for me. 

Known thread in NuGet here: https://github.com/NuGet/Home/issues/1359

It looks like they just released support for 2.0 as of three days ago? So maybe we haven't updated that in electron-builder yet?

Sorry if I'm chasing a tail here, just trying to give Develar a hand as well in figuring it out.

**Comment by develar** [comment]
Will be fixed till end of September. I am on vacation. Sorry for regression. In any case please consider to migrate to NSIS.

**Comment by VerteDinde** [comment]
No worries, @develar ! Have a great vacation, if I can help in anyway, let me know. :) Thanks for making this tool!

**Comment by develar** [comment]
@VerteDinde Nsis Supports Auto update, please see https://www.electron.build/auto-update

**Comment by VerteDinde** [comment]
@develar Don't you need a .nupkg or delta package created with NSIS to make auto update work? Does NSIS create that by default? Mine only creates an .exe currently, so maybe my set up is incorrect if so. Thanks!

**Comment by develar** [comment]
@VerteDinde These files are not required for NSIS.

**Comment by steverandy** [comment]
@develar I understand electron-builder recommends to use nsis, but my project has already chosen squirrel.

I'm looking for a quick solution if there is any.
Will be possible to overwrite the version string that is passed to nuget.

For example by overwriting `1.0.0-beta.24` to `1.0.0-beta24`.
This is actually what electron-builder v19.16.0 does. But for some reason the behavior changes for the version above v19.16.0.

**Comment by develar** [comment]
@steverandy Thanks a lot for the fix, electron-builder-squirrel-windows 19.35.2

**Comment by steverandy** [comment]
@develar no problem. Thanks for making the release quickly!
